### PR TITLE
Rename SmellRepository to DetectorRepository.

### PR DESCRIPTION
**Pull request by troessner**
Fixes [1] of #1103

### CODE REVIEW

**Comment by troessner** [comment]
Ready for review!

File: lib/reek/smells/detector_repository.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ module Smells
     #
     # Contains all the existing smells and exposes operations on them.
```
Comment (mvz) [inline lib/reek/smells/detector_repository.rb]: smells -> smell detectors

**Comment by mvz** [comment]
One small nit & it needs a rebase. Other than that, LGTM!

**Comment by troessner** [comment]
Fixed and rebased!

**Pull request merged**